streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -132,11 +132,9 @@ private void flushAndCheckpointState() {
      * - {@link #commit()}
      * - close state
      * <pre>
-     * @param isZombie ignored by {@code StandbyTask} as it can never be a zombie
      */
     @Override
-    public void close(final boolean clean,
-                      final boolean isZombie) {
+    public void close(final boolean clean) {
```

Comment:
q: about isZombie removal, seems like we just remove the logic of not aborting transaction when the task being closed is treated as zombie, is that correct? So does that mean the old logic is unnecessary?

Reply:
Well it is not unnecessary, it is just that when  I did the refactoring and move it to the record collector, what I did is that we only start txn when there is non inflight, and we commit / abort txn there is one inflight, e.g. as in maybeAbort:

```
if (eosEnabled && transactionInFlight)
```

So when we are closing because of a zombie, the txn may be inflight or not, in which case we would call abortTxn or not, and this logic is wrapped in the `close` call and be agnostic to the task itself.

Does that make sense?

Reply:
Sounds good!